azure-mgmt-trafficmanager/src/main/java/com/microsoft/azure/management/trafficmanager/TrafficManagerEndpoint.java

```diff
@@ -0,0 +1,500 @@
+     * profile definition.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of a traffic manager profile endpoint definition.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface Blank<ParentT> extends WithEndpointType<ParentT> {
+        }
+
+        /**
+         * The stage of the traffic manager profile endpoint definition allowing to specify the type.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithEndpointType<ParentT> {
+            /**
+             * Specify that the endpoint is an Azure cloud service, web app or public ip.
+             *
+             * @param resourceId the Azure resource id
+             * @return the next stage of the endpoint definition
+             */
+            WithAttach<ParentT> withTargetAzureResourceId(String resourceId);
+
+            /**
+             * Specify that the endpoint is an external endpoint that is not hosted in Azure.
+             *
+             * @param externalFqdn the external FQDN
+             * @return the next stage of the endpoint definition
+             */
+            WithSourceTrafficLocation<ParentT> withExternalFqdn(String externalFqdn);
+
+            /**
+             * Specify that the endpoint is a nested traffic manager profile.
+             *
+             * @param nestedProfile the nested traffic manager profile
+             * @return the next stage of the endpoint definition
+             */
+            WithEndpointThreshold<ParentT> withNestedProfile(TrafficManagerProfile nestedProfile);
+        }
+
+        /**
+         * The stage of the nested traffic manager profile endpoint definition allowing to specify the minimum
+         * endpoints to be online in the nested profile to consider it as not degraded.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithEndpointThreshold<ParentT> {
+            /**
+             * Specifies the child endpoint threshold.
+             *
+             * @param count the number of child endpoints to be online to consider nested profile as healthy
+             * @return the next stage of the endpoint definition
+             */
+            WithSourceTrafficLocation<ParentT> withMinimumChildEndpoints(int count);
+        }
+
+        /**
+         * The stage of the traffic manager endpoint definition allowing to specify the location of the external
+         * or nested profile endpoints.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithSourceTrafficLocation<ParentT> {
+            /**
+             * Specifies the location of the endpoint that will be used when the parent profile is configured with
+             * Performance routing method {@link TrafficRoutingMethod#PERFORMANCE}.
```

Comment:
minor suggested polish for clarity and proper capitalization: "Specifies the region of the endpoint that will be used when the performance-based routing method is enabled on the profile."